33-room-occupancy-monitor/README.md

```diff
@@ -1,84 +1,82 @@
-1. Start the program in debug mode (again, how you do this depends on the IDE: VS Code or STM32CubeIDE). In your terminal emulator's output, you should see something like this:
+Note that for `pyboard.py` to work, you'll need to install [pyserial](https://pypi.org/project/pyserial/) with `pip install pyserial`, if you don't have it installed already.
 
-```
-===================
-===================
-===== SPARROW =====
-===================
-Feb 16 2023 22:42:58
-2037335832365003001d001f
-APPLICATION HOST MODE
-CONSOLE TRACE ENABLED
-```
+#### Running `main.py`
 
-You may also see a message like this:
+Before running `main.py`, uncomment this line: `# product_uid = 'com.your-company:your-product-name'`. Replace `com.my-company.my-name:my-project` with the [ProductUID of the Notehub project](https://dev.blues.io/notehub/notehub-walkthrough/#finding-a-productuid) you created in [Notehub Setup](#notehub-setup).
+
+Copy `main.py` over to the board with this command:
 
 ```
-pir: 1 motion events sensed
+python pyboard.py -d <serial port> --no-soft-reset -f cp main.py :/
 ```
 
-At startup, the PIR sensor may detect motion even if there is none. This is a one-off false positive that can be ignored, and the sensor should work as intended thereafter. If you encounter additional false positives, you may want to try increasing the sensor's threshold value, which makes the sensor less sensitive. This can be done by defining the macro `PIR_THRESHOLD_DEFAULT` at build-time (e.g. via `-DCMAKE_C_FLAGS="-DPIR_THRESHOLD_DEFAULT=50"`). If this macro isn't defined by the user, it'll default to 100 in pir.c:
+ Make sure to replace `<serial port>` with your serial port. `main.py` will start running after boot up.
 
-```c
-#ifndef PIR_THRESHOLD_DEFAULT
-#define PIR_THRESHOLD_DEFAULT 100
-#endif
-```
+### Hardware
 
-The value must be between 0 and 255. If you find that the sensor is not sensitive enough, you can try decreasing the threshold value.
+There are two primary pieces of hardware: the magnetic door switch and a PIR motion sensor.
+
+The magnetic door switch consists of two plastic terminals, one of which has a pair of wires coming out of it. When the terminals are brought into contact (or close proximity), an internal [reed switch](https://en.wikipedia.org/wiki/Reed_switch) is closed and the two wires are electrically connected. As shown in the SparkFun product link above, the terminal with the wires is typically mounted to a door frame and the other terminal is mounted to the door such that when the door is closed, the two terminals are adjacent, closing the switch. Since we're just testing things out, don't mount the terminals yet. Instead, keep them on your work surface and plug the wires into the GP21 and GND ports of the Pico (it doesn't matter which wire goes into which port). When the switch is closed, GP21 will connect to GND. Otherwise, GP21 will be driven high by an internal pull-up.
```

Comment:
> As shown in the SparkFun product link above

I’d just make this another link to SparkFun. It’s annoying for the reader to have to scroll up to find this. Or just take the image from SparkFun and include it in the README directly. They won’t care because we’re promoting their product.

> Instead, keep them on your work surface and plug the wires into the GP21 and GND ports of the Pico (it doesn't matter which wire goes into which port).

I feel like this needs a photo.

Reply:
Added a photo now too.